### PR TITLE
Add documentation about shoot networking

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area documentation
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR adds end-user information about Pod networks.

**Special notes for your reviewer**:
/invite @DockToFuture @mvladev

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: docs/usage/shoot_networking.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+nodeCIDRMaskSize: /24
+-------------------------
+
+Number of podCIDRs: 16 --> max. Node count 
+Number of IPs per podCIDRs: 256
+```
+
+With the configuration above a Shoot cluster can at most have **16 nodes** which are ready to run workload in the Pod network.
+
+Beside the configuration in `.spec.networking.pods`, users can change the `nodeCIDRMaskSize` used by Kube-Controller-Manager.
```
Comment (vpnachev) [inline docs/usage/shoot_networking.md]:
```suggestion
Beside the configuration in `.spec.networking.pods`, users can tune the `nodeCIDRMaskSize` used by Kube-Controller-Manager on shoot creation.
```

**Review by vpnachev — COMMENTED** [review]
Looks good to me. I have just one suggestion w.r.t. immutability of `nodeCIDRMaskSize`.

File: docs/usage/shoot_networking.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Shoot Networking
+
+This document contains network related information for Shoot clusters.
+
+## Pod Network
+
+A Pod network is imperative for any kind of cluster communication with Pods not started within the Node's host network.
+More information about the Kubernetes network model can be found [here](https://kubernetes.io/docs/concepts/cluster-administration/networking/).
+
+Gardener allows users to configure the Pod network's IP configuration during Shoot creation:
```
Comment (rfranzke) [inline docs/usage/shoot_networking.md]:
```suggestion
Gardener allows users to configure the Pod network's CIDR during Shoot creation:
```

File: docs/usage/shoot_networking.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  networking:
+     type: <some-network-extension-name> # {calico,cilium}
+     pods: 100.96.0.0/16
+     nodes: ...
+     services: ...
```
Comment (rfranzke) [inline docs/usage/shoot_networking.md]:
```suggestion
    type: <some-network-extension-name> # {calico,cilium}
    pods: 100.96.0.0/16
    nodes: ...
    services: ...
```

**Comment by timuthy** [comment]
Thanks for your reviews @vpnachev and @rfranzke. I've considered your feedback and adapted the PR.

**Review by rfranzke — APPROVED** [review]
/lgtm
/needs second-opinion

**Review by DockToFuture — APPROVED** [review]
It's well written to explain the general concepts behind `IPAM` in k8s.

In our setup we use `Calico` as CNI plugin which has a similar but more dynamic approach of assigning IPs from `ippools` to the pods. The default block size is `/26` and a node can get multiple chunks assigned from the podCidr split up. If you want you can also mention this for calico but the administrator/operator can also dig that up himself. I guess it's to much to explain all that concepts especially since cilium is supporting multiple `IPAM` modes.

**Review by vpnachev — APPROVED** [review]
/lgtm
/ok-to-test

**Comment by timuthy** [comment]
> In our setup we use `Calico` as CNI plugin which has a similar but more dynamic approach of assigning IPs from `ippools` to the pods. The default block size is `/26` and a node can get multiple chunks assigned from the podCidr split up. If you want you can also mention this for calico but the administrator/operator can also dig that up himself. I guess it's to much to explain all that concepts especially since cilium is supporting multiple `IPAM` modes.

Thanks @DockToFuture, I think it's very important for the user to know. However, the Calico extension repo might be a better place to put such information. If we put it here, I fear that it got outdated unnoticed because one working on the Calico extension rather doesn't have the documentation in `gardener/gardener` in mind. WDYT?

**Comment by DockToFuture** [comment]
Yes, I agree.

**Pull request merged**